### ISSUE TITLE
Extracting fails due to TypeError: Cannot read property 'kind' of undefined

### ISSUE THREAD

**Issue by eaardal** [formatjs/formatjs#3208]
Hi

We're using @formatjs/cli in a NextJS/React app to extract i18n strings using the script 

```
formatjs extract './src/{pages,components,sections}/**/*.{js,ts,tsx}' --format simple --id-interpolation-pattern '[sha512:contenthash:base64:6]' --out-file ./src/content/locales/en.json --ignore **/_document.page.tsx **/*.test.* **/*.stories.*
```

However, some statements in some files will cause the extractor to fail on that file and none of the i18n strings in the failing file will be extracted. It always fails with a warning like this:

```
[@formatjs/cli] [WARN] TypeError: Error processing file ./src/components/AppBarUserMenu.tsx 
Cannot read property 'kind' of undefined
```

Adding `--throws` to the extract command it returns this stack:

```
TypeError: Error processing file ./src/components/AppBarUserMenu.tsx 
Cannot read property 'kind' of undefined
    at Object.isObjectLiteralExpression (/Users/eaardal/dev/git/tv2/rp-ad-web/node_modules/typescript/lib/typescript.js:28029:21)
    at extractMessagesFromCallExpression (/Users/eaardal/dev/git/tv2/rp-ad-web/node_modules/@formatjs/ts-transformer/src/transform.js:380:16)
    at visitor (/Users/eaardal/dev/git/tv2/rp-ad-web/node_modules/@formatjs/ts-transformer/src/transform.js:405:15)
    at visitNode (/Users/eaardal/dev/git/tv2/rp-ad-web/node_modules/typescript/lib/typescript.js:85158:23)
    at Object.visitEachChild (/Users/eaardal/dev/git/tv2/rp-ad-web/node_modules/typescript/lib/typescript.js:85622:64)
    at visitor (/Users/eaardal/dev/git/tv2/rp-ad-web/node_modules/@formatjs/ts-transformer/src/transform.js:409:19)
    at visitNodes (/Users/eaardal/dev/git/tv2/rp-ad-web/node_modules/typescript/lib/typescript.js:85211:48)
    at Object.visitEachChild (/Users/eaardal/dev/git/tv2/rp-ad-web/node_modules/typescript/lib/typescript.js:85616:50)
    at visitor (/Users/eaardal/dev/git/tv2/rp-ad-web/node_modules/@formatjs/ts-transformer/src/transform.js:409:19)
    at visitNode (/Users/eaardal/dev/git/tv2/rp-ad-web/node_modules/typescript/lib/typescript.js:85158:23)
```

Which I narrowed down to `descriptorsObj` being undefined when passed into `ts.isObjectLiteralExpression()` here: https://github.com/formatjs/formatjs/blob/82a2e5d07f9a999624ffb6387a58464f9a5f7399/packages/ts-transformer/src/transform.ts#L619

I haven't found any pattern in our code when or when not this happens, it has happend four times with four separate statements in four different files, all looked different. In this particular example the failing line was a method call in a callback function to a react button handler where the method call was to a static function on a utility class. Nothing special about the React component, the callback handler or the utility class that we can figure out.  The failing line has nothing to do with i18n.

All we're able to conclude is that adding a sanity check to the marked line such as the following solves the problem and enables extraction to run uninterrupted:
```
if (descriptorsObj && ts.isObjectLiteralExpression(descriptorsObj)) {
```

**Comment by longlho** [comment]
can you share the content of the file? What TS version are you using?

**Comment by eaardal** [comment]
The file (./src/components/AppBarUserMenu.tsx). Line with `LocalStorageUtil.removeBearerToken()` causes the error:

```jsx
import React, { useState } from 'react'
import styled from '@emotion/styled'
import { Avatar, Tooltip } from '@mui/material'
import IconButton from '@mui/material/IconButton'
import MenuItem from '@mui/material/MenuItem'
import MuiMenu from '@mui/material/Menu'
import { Typography } from '@mui/material'
import { FormattedMessage } from 'react-intl'
import { useAuth0 } from '@auth0/auth0-react'
import { useRouter } from 'next/router'
import RoutingUtil, { AppRoute } from '@app/utils/RoutingUtil'
import { getConfig } from '@app/infrastructure/config'
import { TV2ThemeProps } from '@app/styles/tv2theme'
import LocalStorageUtil from '@app/utils/LocalStorageUtil'

const LetterAvatar = styled(Avatar)`
  background-color: ${(props: TV2ThemeProps) => props.theme!.tv2.colors.primary};
  color: ${(props: TV2ThemeProps) =>
    props.theme!.palette.getContrastText(props.theme!.tv2.colors.primary)};
`

function AppBarUserMenu(): JSX.Element {
  const { user, logout } = useAuth0()
  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null)
  const open = Boolean(anchorEl)
  const router = useRouter()
  const config = getConfig()

  function handleMenu(event: React.MouseEvent<HTMLElement>) {
    setAnchorEl(event.currentTarget)
  }

  function handleClose() {
    setAnchorEl(null)
  }

  async function goToProfile(): Promise<void> {
    handleClose()
    return RoutingUtil.push(AppRoute.UserProfile, router)
  }

  function logOut() {
    handleClose()
    LocalStorageUtil.removeBearerToken()
    logout({
      returnTo: window.location.origin,
      client_id: config.auth0.clientId,
    })
  }

  if (typeof user === 'undefined') {
    return <div />
  }

  return (
    <>
      <IconButton
        aria-label="account of current user"
        aria-controls="menu-appbar"
        aria-haspopup="true"
        onClick={handleMenu}
        color="inherit"
        size="large"
      >
        <Tooltip
          placement="left"
          title={
            <p>
              {user.name || 'Unknown name'}
              <br />
              {user.email || 'Unknown email'}
            </p>
          }
        >
          {user.picture ? (
            <Avatar alt={`profile picture for ${user.name || 'UNKNOWN'}`} src={user.picture} />
          ) : (
            <LetterAvatar>{user.name?.substring(0, 1) || '?'}</LetterAvatar>
          )}
        </Tooltip>
      </IconButton>
      <MuiMenu
        id="menu-appbar"
        anchorEl={anchorEl}
        anchorOrigin={{
          vertical: 'top',
          horizontal: 'right',
        }}
        keepMounted
        transformOrigin={{
          vertical: 'top',
          horizontal: 'right',
        }}
        open={open}
        onClose={handleClose}
      >
        <MenuItem onClick={goToProfile}>
          <Typography>
            <FormattedMessage id="appBar.loggedInMenu.userProfileButton" defaultMessage="Profile" />
          </Typography>
        </MenuItem>
        <MenuItem onClick={logOut}>
          <Typography>
            <FormattedMessage id="appBar.loggedInMenu.logOutButton" defaultMessage="Log out" />
          </Typography>
        </MenuItem>
      </MuiMenu>
    </>
  )
}

export default AppBarUserMenu
```

TypeScript for the project is resolved to 4.4.4. We don't have an explicit dependency on TS but get it through NextJS. If i put `"ts-version": "tsc --version"` in package.json and run `npm run ts-version` it returns 4.4.4. My machine has 4.3.5 installed elsewhere.

We use @formatjs/cli 4.3.1

Just now I started to simplify the `formatjs extract` CLI command as I wanted to run the command on only the problem file (as I suspect you would do) and discovered that the error disappeared when I removed `--additional-component-names`. 

After tinkering with it a bit, I narrowed it down to spacing between list elements.

This produces the error:
```
--additional-component-names ['Foo', 'Bar']"
```

This does not:
```
--additional-component-names ['Foo','Bar']"
```

Difference being the space between `'Foo'` and `'Bar'`

The complete command (`npm run extract-tmp`):
```
"extract-tmp": "formatjs extract './src/components/AppBarUserMenu.tsx' --format simple --id-interpolation-pattern '[sha512:contenthash:base64:6]' --out-file ./src/content/locales/en.tmp.json --additional-component-names ['Foo','Bar']",
```

Hope this helps :)

**Comment by longlho** [comment]
hmm it's supposed to be a comma-separated list so `--additional-component-names="Foo,Bar"` is the right one. We use `commander.js` to parse the opt so this might be an issue upstream in that library. The space definitely messes up the args. The arg is not really JSON format. I've fixed the help text and hopefully that's clearer.
